### PR TITLE
Deprecate `cardAddChallenge` & add internal property

### PR DESCRIPTION
**Pull request by scannillo**
Follow up to PR #1122 & [this comment](https://github.com/braintree/braintree_ios/pull/1122#discussion_r1376494256)

### Summary of changes

- Deprecate `BTThreeDSecure.cardAddChallenge`
- Create internal variable for `cardAddChallenge` to use within the SDK
    - Since we cannot silence deprecation warnings in Swift, Jax's PR above revealed that referencing deprecated methods & properties internally will surface errors to merchants
    - This works around that limitation by only accessing the _internal_ variable for `cardAddChallenge` throughout the SDK, while still deprecating the merchant-facing public property
    - Since the internal property is not deprecated, our SDK will not trigger warnings

### Checklist

- [X] Added a changelog entry

### Authors
@scannillo

### CODE REVIEW

File: Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift [hunk 1]
```diff
@@ -58,8 +58,16 @@ import BraintreeCore
     /// If set to `.notRequested` the authentication challenge will not be requested from the issuer.
     /// If set to `.unspecified`, when the amount is 0, the authentication challenge will be requested from the issuer.
     /// If set to `.unspecified`, when the amount is greater than 0, the authentication challenge will not be requested from the issuer.
-    public var cardAddChallenge: BTThreeDSecureCardAddChallenge = .unspecified
-
+    @available(*, deprecated, message: "Use `cardAddChallengeRequested` instead")
```
Comment (jaxdesmarais) [inline Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift:61]: I think we can use:
```swift
@available(*, deprecated, renamed: "cardAddChallengeRequested" message: "whatever we want for the message here")
```
I believe this way lets you use `Fix` in Xcode.

File: Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift [hunk 2]
```diff
@@ -58,8 +58,16 @@ import BraintreeCore
...
+    public var cardAddChallenge: BTThreeDSecureCardAddChallenge {
+        get { return _cardAddChallenge }
```
Comment (jaxdesmarais) [inline Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift:63]:
```suggestion
        get { _cardAddChallenge }
```

**Review by jaxdesmarais — APPROVED** [review]

File: Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift (hunk 1 above)
Comment (scannillo) [reply Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift]: Ooo that's awesome, fixed!

File: Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift (hunk 2 above)
Comment (scannillo) [reply Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift]: Fixed!

**Review by sarahkoop — APPROVED** [review]

**Pull request merged**